src/ethereum/forks/amsterdam/fork.py

```diff
@@ -1129,20 +1151,20 @@ def increase_recipient_balance(recipient: Account) -> None:
 
         new_balance = get_account(block_env.state, wd.address).balance
         track_balance_change(
-            block_env.block_state_changes, wd.address, new_balance
+            block_env.block_state_changes,
+            wd.address,
+            new_balance,
+            block_access_index,
         )
 
         if account_exists_and_is_empty(block_env.state, wd.address):
             destroy_account(block_env.state, wd.address)
 
     # EIP-7928: Normalize balance changes after all withdrawals
     # Filters out net-zero changes
-
     normalize_balance_changes_for_transaction(
         block_env.block_state_changes,
-        BlockAccessIndex(
-            get_block_access_index(block_env.block_state_changes)
-        ),
+        block_access_index,
```

Comment:
If we intend to call this function beyond a transaction processing, might be best to rename it to `normalize_balance_changes` instead

Reply:
I think we can normalize all net-zero filtering here once we remove that from `merge_on_success` as I mentioned before. And yeah, agreed, this is called on withdrawals too so the naming should be better here.